components/containerd/pod/component.go

```diff
@@ -3,113 +3,313 @@ package pod
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"strings"
+	"sync"
 	"time"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+
 	"github.com/leptonai/gpud/components"
 	containerd_pod_id "github.com/leptonai/gpud/components/containerd/pod/id"
 	"github.com/leptonai/gpud/pkg/log"
-	"github.com/leptonai/gpud/pkg/query"
-
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
-func New(ctx context.Context, cfg Config) components.Component {
-	cfg.Query.SetDefaultsIfNotSet()
-	setDefaultPoller(cfg)
-
-	cctx, ccancel := context.WithCancel(ctx)
-	getDefaultPoller().Start(cctx, cfg.Query, containerd_pod_id.Name)
-
-	return &component{
-		rootCtx: ctx,
-		cancel:  ccancel,
-		poller:  getDefaultPoller(),
+func New(ctx context.Context) components.Component {
+	_, cancel := context.WithCancel(ctx)
+	c := &component{
+		rootCtx:  ctx,
+		cancel:   cancel,
+		endpoint: defaultContainerRuntimeEndpoint,
 	}
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
```

Comment:
The first time we have to wait for 1 minute. perhaps we should run checkOnce, then wait for ticker?